server/src/main/java/org/elasticsearch/repositories/RepositoriesStats.java

```diff
@@ -26,52 +27,115 @@
-            repositoryThrottlingStats = in.readMap(ThrottlingStats::new);
+            repositorySnapshotStats = in.readMap(SnapshotStats::readFrom);
         } else {
-            repositoryThrottlingStats = new HashMap<>();
+            repositorySnapshotStats = new HashMap<>();
         }
     }
 
-    public RepositoriesStats(Map<String, ThrottlingStats> repositoryThrottlingStats) {
-        this.repositoryThrottlingStats = new HashMap<>(repositoryThrottlingStats);
+    public RepositoriesStats(Map<String, SnapshotStats> repositorySnapshotStats) {
+        this.repositorySnapshotStats = new HashMap<>(repositorySnapshotStats);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         if (out.getTransportVersion().onOrAfter(TransportVersions.V_8_9_X)) {
-            out.writeMap(repositoryThrottlingStats, StreamOutput::writeWriteable);
+            out.writeMap(repositorySnapshotStats, StreamOutput::writeWriteable);
         }
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.field("repositories", repositoryThrottlingStats);
+        builder.field("repositories", repositorySnapshotStats);
         return builder;
     }
 
-    public Map<String, ThrottlingStats> getRepositoryThrottlingStats() {
-        return Collections.unmodifiableMap(repositoryThrottlingStats);
+    public Map<String, SnapshotStats> getRepositorySnapshotStats() {
+        return Collections.unmodifiableMap(repositorySnapshotStats);
     }
 
-    public record ThrottlingStats(long totalReadThrottledNanos, long totalWriteThrottledNanos) implements ToXContentObject, Writeable {
+    public record SnapshotStats(
+        long shardSnapshotsStarted,
+        long shardSnapshotsCompleted,
+        long shardSnapshotsInProgress,
+        long totalReadThrottledNanos,
+        long totalWriteThrottledNanos,
+        long numberOfBlobsUploaded,
+        long numberOfBytesUploaded,
+        long totalUploadTimeInNanos,
+        long totalUploadReadTimeInNanos
+    ) implements ToXContentObject, Writeable {
 
-        ThrottlingStats(StreamInput in) throws IOException {
-            this(in.readVLong(), in.readVLong());
+        public static SnapshotStats readFrom(StreamInput in) throws IOException {
+            final long totalReadThrottledNanos = in.readVLong();
+            final long totalWriteThrottledNanos = in.readVLong();
+            if (in.getTransportVersion().onOrAfter(TransportVersions.EXTENDED_SNAPSHOT_STATS_IN_NODE_INFO)) {
+                return new SnapshotStats(
+                    in.readLong(),
+                    in.readLong(),
+                    in.readLong(),
+                    totalReadThrottledNanos,
+                    totalWriteThrottledNanos,
+                    in.readLong(),
+                    in.readLong(),
+                    in.readLong(),
+                    in.readLong()
```

Comment:
I'd expect these to be represented using `VLong` rather than a bare `Long` - mostly they're going to be quite close to zero. Tho that means we can't just use `-1` to mean "missing". Maybe we should also include a boolean up front to indicate that we only include the legacy throttling stats?